api/main.py

```diff
@@ -94,8 +94,24 @@ async def get_nodes(request: Request):
 async def post_node(node: Node, token: str = Depends(get_user)):
     try:
         if node.id is None:
-            obj = await db.create(node)
-            op = 'created'
+            # Check if node exists or not
+            find_node = await db.find_one(
```

Comment:
The issue here is that there is still a potential race condition between doing the `db.find_one()` call and the `db.create()` one.  So maybe we need to rely on some Mongo DB feature to specify that documents need to be unique with some fields, which would be described in the model.  I see it's possible to do that using an index:
https://docs.mongodb.com/manual/tutorial/unique-constraints-on-arbitrary-fields/

I haven't looked at how to do that with Pydantic models but hopefully there's a simple solution for it.  Maybe this can be removed from this PR and added as a follow-up one: first we'll have a solution to differentiate POST and PUT requests and assume there aren't any race condition issues.  Then we can look into having a way to guarantee some Mongo DB documents don't have duplicates.

Reply:
I think we should create a separate issue for this.

Reply:
Yes, so for this PR we can just drop the code that looks for an existing node with the same attributes.

Reply:
Then I need to drop the two commits from the PR https://github.com/kernelci/kernelci-api/pull/86/commits/55f71a99da736fe3eb322be54e18c326200699d5 and https://github.com/kernelci/kernelci-api/pull/86/commits/c14e6db6cb49207bb87183ad6cb93d9c25e336fa.